service/src/main/java/org/cbioportal/service/impl/CoExpressionServiceImpl.java

```diff
@@ -117,31 +109,49 @@
         return computedCoExpressions;
     }
 
+    @Override
+    public List<CoExpression> getCoExpressions(String molecularProfileId, String sampleListId, String geneticEntityId,
+                                               CoExpression.GeneticEntityType geneticEntityType, Double threshold)
+        throws MolecularProfileNotFoundException, GenesetNotFoundException, GeneNotFoundException {
+
+        List<String> sampleIds = sampleListRepository.getAllSampleIdsInSampleList(sampleListId);
+        if (sampleIds.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        return fetchCoExpressions(molecularProfileId, sampleIds, geneticEntityId, geneticEntityType, threshold);
+    }
+
     @Override
     public List<CoExpression> fetchCoExpressions(String molecularProfileId, List<String> sampleIds, 
                                                  String queryGeneticEntityId, CoExpression.GeneticEntityType geneticEntityType, 
                                                  Double threshold)
         throws MolecularProfileNotFoundException, GenesetNotFoundException, GeneNotFoundException {
-        
-        List<? extends MolecularAlteration> molecularAlterations = null;
+
+        // For the purpose of the CoExpression computation, we separate the MolecularAlteration
+        // (genetic_alteration table record) for the query gene/geneset from the genetic alterations
+        // for the remaining genes/geneset in the profile.
+        MolecularAlteration queryMolecularDataList = null;
+        Iterable<? extends MolecularAlteration> maItr = null;
         if (geneticEntityType.equals(GeneticEntityType.GENE)) {
-            molecularAlterations = molecularDataService.getMolecularAlterations(
-                molecularProfileId, null, "SUMMARY");
+            List<Integer> queryGeneticEntityIds = Arrays.asList(Integer.valueOf(queryGeneticEntityId));
+            maItr = molecularDataService.getMolecularAlterations(molecularProfileId, queryGeneticEntityIds, "SUMMARY");
         } else if (geneticEntityType.equals(GeneticEntityType.GENESET)) {
-            molecularAlterations = genesetDataService.getGenesetAlterations(
-                molecularProfileId, null);
+            List<String> queryGeneticEntityIds = Arrays.asList(queryGeneticEntityId);
+            maItr = genesetDataService.getGenesetAlterations(molecularProfileId, queryGeneticEntityIds);
+        }
+        for (MolecularAlteration ma : maItr) {
+            queryMolecularDataList = ma;
```

Comment:
should this be adding to this list or since it's just querying one genetic entity id that `queryMoleuclarDataList` is just being reassigned to whatever the first element of `maItr` is (assuming it's non-empty)

Reply:
Basically yes - its only passing in one entity id so it only expects on record back.

Reply:
thanks for clarifying, just wanted to confirm